scripts/zkai_attention_derived_d128_native_mlp_proof_route_gate.py

```
@@ -0,0 +1,670 @@
+    current_input = loaded[CURRENT_MLP_FUSED_INPUT]
+    current_envelope = loaded[CURRENT_MLP_FUSED_ENVELOPE]
+    derived_input_commitment = _commitment(
+        chain_summary.get("derived_input_activation_commitment"),
+        "derived input activation commitment",
+    )
+    current_input_commitment = _commitment(
+        current_input.get("input_activation_commitment"),
+        "current MLP fused input activation commitment",
+    )
+    if derived_input_commitment != EXPECTED_DERIVED_INPUT_COMMITMENT:
+        raise NativeMlpProofRouteError("derived input commitment drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)